### ISSUE TITLE
Issue: when using the map_rerank & refine, occur the " list index out of range" (already modify the llm.py file, but only map_reduce can work well)

### ISSUE THREAD

**Issue by jason850124** [langchain-ai/langchain#9111]
### Issue you'd like to raise.

I just tried the simple example with the latest version of langchain, but I meet the problem with " list index out of the range", when I modify the chain_type in "RetrievalQA.from_chain_type" function, the error messages are as follows :

map_rerank :
-----------------------------------------------------------------------------------------------------------------------------
File ~\anaconda3\envs\pytorch\Lib\site-packages\langchain\chains\combine_documents\map_rerank.py:194, in MapRerankDocumentsChain._process_results(self, docs, results)
    190 typed_results = cast(List[dict], results)
    191 sorted_res = sorted(
    192     zip(typed_results, docs), key=lambda x: -int(x[0][self.rank_key])
    193 )
--> 194 output, document = sorted_res[0]
    195 extra_info = {}
    196 if self.metadata_keys is not None:
IndexError: list index out of range
-----------------------------------------------------------------------------------------------------------------------------

refine:
-----------------------------------------------------------------------------------------------------------------------------
File ~\anaconda3\envs\pytorch\Lib\site-packages\langchain\chains\combine_documents\refine.py:203, in RefineDocumentsChain._construct_initial_inputs(self, docs, **kwargs)
    200 def _construct_initial_inputs(
    201     self, docs: List[Document], **kwargs: Any
    202 ) -> Dict[str, Any]:
--> 203     base_info = {"page_content": docs[0].page_content}
    204     base_info.update(docs[0].metadata)
    205     document_info = {k: base_info[k] for k in self.document_prompt.input_variables}
IndexError: list index out of range
-----------------------------------------------------------------------------------------------------------------------------

Thanks for your help!

**Comment by jason850124** [comment]
Hi! Dosu,

Thanks for your reply, but I don't think is not the problem you mentioned, because I modified the llm.py file by following the solution #5769, and used map_reduce, which can well, and then I just changed the "map_reduce" to "map_rerank" (or "refine") which can accept the same data type, then this issue will occur. 

So I think maybe something is wrong and cause the combine_docs or acombine_docs to get nothing.

BS,
Jason
